cmd/collector/app/span_processor.go

```diff
@@ -178,10 +178,21 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 }
 
 func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	// TODO add support for deduping tags, https://github.com/jaegertracing/jaeger/issues/1778
+	dedupKey := make(map[string]struct{})
+	for _, tag := range span.Process.Tags {
+		if value, ok := sp.collectorTags[tag.Key]; ok && value == tag.AsString() {
+			sp.logger.Debug("ignore collector process tags", zap.String("key", tag.Key), zap.String("value", value))
+			dedupKey[tag.Key] = struct{}{}
+		}
+	}
+	// ignore collector tags if has the same key-value in spans
 	for k, v := range sp.collectorTags {
-		span.Process.Tags = append(span.Process.Tags, model.String(k, v))
+		if _, ok := dedupKey[k]; !ok {
+			span.Process.Tags = append(span.Process.Tags, model.String(k, v))
+		}
 	}
+	typedTags := model.KeyValues(span.Process.Tags)
+	typedTags.Sort()
```

Comment:
why is sort necessary?

Reply:
I just follow `NewProcess` function. And make tags sorted after collector processor. https://github.com/jaegertracing/jaeger/blob/a7be3a8e527738e23c14e4e314b917c5d485f5bd/model/process.go#L28

Reply:
Maybe related issues: https://github.com/jaegertracing/jaeger/issues/713

Reply:
I'm not familiar about how to address dedup tag in UI.
@yurishkuro Any comment for that?

Reply:
ok, let's keep the sort